sql/rpl_slave.cc

```diff
@@ -9736,6 +9736,27 @@ static int check_slave_sql_config_conflict(const Relay_log_info *rli) {
   return 0;
 }
 
+// This function will generate the string containing current master host and
+// port info if available.
+void get_active_master_info(std::string &str_ptr) noexcept {
```

Comment:
Return std::string

Reply:
I kept the original solution because `std::string extra_info = get_active_master_info();` would require an additional string copy.

Reply:
It should not: RVO